GeneralsMD/CMakeLists.txt

```diff
@@ -12,5 +12,17 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 # Add main build targets
 add_subdirectory(Code)
 
+# If we are building on windows for windows, try and get the game install path from the registry.
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows" AND CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    get_filename_component(ZEROHOUR_INSTALL_PREFIX  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Electronic Arts\\EA Games\\Command and Conquer Generals Zero Hour;InstallPath]" ABSOLUTE CACHE)
+endif()
+
+if(ZEROHOUR_INSTALL_PREFIX AND NOT ZEROHOUR_INSTALL_PREFIX STREQUAL "/registry")
+    install(TARGETS z_generals RUNTIME DESTINATION "${ZEROHOUR_INSTALL_PREFIX}")
+    install(FILES $<TARGET_PDB_FILE:z_generals> DESTINATION "${ZEROHOUR_INSTALL_PREFIX}" OPTIONAL)
+    install(TARGETS z_worldbuilder RUNTIME DESTINATION "${ZEROHOUR_INSTALL_PREFIX}" OPTIONAL)
+    install(FILES $<TARGET_PDB_FILE:z_worldbuilder> DESTINATION "${ZEROHOUR_INSTALL_PREFIX}" OPTIONAL)
```

Comment:
Is the INSTALL target enabled by default for MSVC presets, or remembered on cmake reconfigure? If not, can we do that?

It was a source of frustration with Thyme, because INSTALL kept getting disabled on cmake reconfigure and I ended up testing outdated things more than once.

Reply:
After cmake reconfigure, I go to Target Configuration in Visual Studio, put the checkbox next to INSTALL, ok, and then INSTALL will run on any regular build (F7).

It's only after cmake reconfigure, that this is erased again.

Reply:
I think that annoyance will persist unless/until the built in cmake handling can handle CMakePresets.json schema versions that can build a workflow with build and install presets. CMake regenerates the solution file, it doesn't know anything about changes you have made to it to try and persist them.